packages/integration-sdk-testing/src/context.ts

```diff
@@ -92,32 +110,45 @@ export function createMockExecutionContext<
 }
 
 function isOptionsWithInstanceConfig<
-  TConfig extends IntegrationInstanceConfig = IntegrationInstanceConfig,
+  TInstanceConfig extends IntegrationInstanceConfig = IntegrationInstanceConfig,
 >(
-  options: CreateMockExecutionContextOptions<TConfig>,
-): options is CreateMockExecutionContextOptionsWithInstanceConfig<TConfig> {
+  options: CreateMockExecutionContextOptions<TInstanceConfig>,
+): options is CreateMockExecutionContextOptionsWithConfigs<TInstanceConfig> {
   return !!(options as any).instanceConfig;
 }
 
 export type CreateMockStepExecutionContextOptions<
-  TConfig extends IntegrationInstanceConfig = IntegrationInstanceConfig,
-> = CreateMockExecutionContextOptions<TConfig> & CreateMockJobStateOptions;
+  TInstanceConfig extends IntegrationInstanceConfig = IntegrationInstanceConfig,
+  TExecutionConfig extends IntegrationExecutionConfig = IntegrationExecutionConfig,
+> = CreateMockExecutionContextOptions<TInstanceConfig, TExecutionConfig> &
+  CreateMockJobStateOptions & {
+    stepMetadata: StepMetadata;
```

Comment:
This looks like a breaking change too. `stepMetadata` is now enforced. `createMockStepExecutionContext` is used in many integrations: https://github.com/search?q=org%3AJupiterOne+%22createMockStepExecutionContext%22&type=code

I think we will want something like this:

```ts
export type CreateMockStepExecutionContextOptions<
  TInstanceConfig extends IntegrationInstanceConfig = IntegrationInstanceConfig,
  TExecutionConfig extends IntegrationExecutionConfig = IntegrationExecutionConfig,
> = CreateMockExecutionContextOptions<TInstanceConfig, TExecutionConfig> &
  CreateMockJobStateOptions & {
    stepMetadata?: StepMetadata;
  };

// ...
// ...

export function createMockStepExecutionContext<
  TInstanceConfig extends IntegrationInstanceConfig = IntegrationInstanceConfig,
  TExecutionConfig extends IntegrationExecutionConfig = IntegrationExecutionConfig,
>(
  options: CreateMockStepExecutionContextOptions<
    TInstanceConfig,
    TExecutionConfig
  > = {
    instanceConfigFields:
      {} as IntegrationInstanceConfigFieldMap<TInstanceConfig>,
  },
): MockIntegrationStepExecutionContext<TInstanceConfig, TExecutionConfig> {
  return {
    ...createMockExecutionContext<TInstanceConfig, TExecutionConfig>(options),
    stepMetadata: options.stepMetadata || ({} as StepMetadata),
    jobState: createMockJobState(options),
  };
}
```